hazelcast-hibernate53/src/main/java/com/hazelcast/hibernate/PhoneHomeService.java

```diff
@@ -91,25 +92,25 @@ private void send() {
         int retryCount = RETRY_COUNT;
         boolean succeed = false;
         while (retryCount-- > 0 && !succeed) {
-            InputStream in = null;
-            try {
-                URL url = new URL(baseUrl + phoneHomeInfo.getQueryString());
-                URLConnection conn = url.openConnection();
-                conn.setRequestProperty("User-Agent", "Mozilla/5.0");
-                conn.setConnectTimeout((int) TIMEOUT.toMillis());
-                conn.setReadTimeout((int) TIMEOUT.toMillis());
-                in = new BufferedInputStream(conn.getInputStream());
+            try (InputStream ignored = new BufferedInputStream(getUrlConnection().getInputStream())) {
                 succeed = true;
             } catch (Exception e) {
                 if (logger.isFineEnabled()) {
                     logger.fine("Failed to establish home phone call. Retries left: " + retryCount, e);
                 }
-            } finally {
-                closeResource(in);
```

Comment:
This was the cause of the failing compatibility tests, we've got an exception here:

```
Throwable: java.lang.NoSuchMethodError: 'void com.hazelcast.internal.nio.IOUtil.closeResource(java.lang.AutoCloseable)'
```

@TomaszGaweda followed my suggestion and changed `closeResource` signature: https://github.com/hazelcast/hazelcast/pull/23880#discussion_r1129896338

I've changed the implementation to avoid this method and use try-with-resources